tests/discovery/processing/test_discovery_service.py

```diff
@@ -212,6 +212,31 @@ def mock_save_tsne_plot_data(tsne_plot_data):
     assert mock_save_tsne_plot_data_called_with == 'Saving TSNE Plot Data'
 
 
+def test_run_experiment_with_partially_labelled_data(monkeypatch):
```

Comment:
It is not directly a part of this story, but since we are adding more tests for the run_experiment part of the discovery_service, we could think about creating a dedicated testfile, e.g. test_discovery_service_experiment.py and move all tests related to the experiment to that file. This way the test files are a little more comprehensible.

Reply:
Done